pkg/common/metrics/stats_reporter.go

```diff
@@ -0,0 +1,207 @@
+		return ctx
+	}
+
+	newView := &view.View{
+		Name:        name,
+		Description: description,
+		Measure:     measure,
+		Aggregation: view.LastValue(),
+	}
+	err = view.Register(newView)
+	if err != nil {
+		r.logger.Error("failed to register opencensus views", zap.Error(err))
+		return ctx
+	}
+	r.views[name] = newView
+	return ctx
+}
+
+// Record a measurement to the metrics backend, creating a new OpenCensus view if this is a new measurement
+func (r *Reporter) recordMeasurement(metricKey string, saramaKey string, value interface{}) {
+
+	name := fmt.Sprintf("%s.%s", metricKey, saramaKey)
+	description := getDescription(metricKey, saramaKey)
+
+	// Type-switches don't support "fallthrough" so each individual possible type must have its own
+	// somewhat-redundant code block
+	switch value := value.(type) {
+	case int:     r.recordInt(int64(value), name, description)
+	case int8:    r.recordInt(int64(value), name, description)
+	case int16:   r.recordInt(int64(value), name, description)
+	case int32:   r.recordInt(int64(value), name, description)
+	case int64:   r.recordInt(value, name, description)
+	case uint:    r.recordInt(int64(value), name, description)
+	case uint8:   r.recordInt(int64(value), name, description)
+	case uint16:  r.recordInt(int64(value), name, description)
+	case uint32:  r.recordInt(int64(value), name, description)
+	case uint64:  r.recordInt(int64(value), name, description)
+	case float64: r.recordFloat(value, name, description)
+	case float32: r.recordFloat(float64(value), name, description)
+	default:
+		r.logger.Warn("Could not interpret measurement as a number", zap.Any("Sarama Value", value))
+	}
+}
+
+// Record a measurement that is represented by an int64 value
+func (r *Reporter) recordInt(value int64, name string, description string) {
+	measure := stats.Int64(name, description, stats.UnitDimensionless)
+	RecordWrapper(r.tagCtx, measure.M(value))
+	r.createViewIfNecessary(measure, name, description)
+}
+
+// Record a measurement that is represented by a float64 value
+func (r *Reporter) recordFloat(value float64, name string, description string) {
+	measure := stats.Float64(name, description, stats.UnitDimensionless)
+	RecordWrapper(r.tagCtx, measure.M(value))
+	r.createViewIfNecessary(measure, name, description)
+}
+
+// Adds a view for this measurement to this Reporter's map of known views, if not already present
+func (r *Reporter) createViewIfNecessary(measure stats.Measure, name string, description string) {
+	if _, ok := r.views[name]; !ok {
+		r.tagCtx = r.createView(r.tagCtx, measure, name, description)
+	}
+}
+
+// Returns pretty descriptions for known Sarama metrics
+func getDescription(main string, sub string) string {
+	// Run through the list of known replacements that should be made (multiple replacements may happen)
+	for _, replacement := range regexDescriptions {
+		main = replacement.Search.ReplaceAllString(main, replacement.Replace)
+	}
+	return main + ": " + getSubDescription(sub)
+}
```

Comment:
It can't be a straight string-map swap because there are numbers and text in some of the metrics that aren't constant, like the name of the topic or the index of the broker.  Originally I used regex because of those same values (searching for "([9-0]+)" and replacing with "${1}" and similar, but I realized I could just replace the string "up to the value" and not require regex shenanigans.  That means it could be a normal strings.Replace() instead of a more complicated regex.  I'll do a quick test and see if that's actually faster.